packages/xstate-vue/src/useFsm.ts

```diff
@@ -0,0 +1,24 @@
+import { ref, onBeforeUnmount, Ref } from '@vue/composition-api';
+import { StateMachine, EventObject, interpret } from '@xstate/fsm';
+import { AnyEventObject } from 'xstate';
+
+export function useFsm<TC, TE extends EventObject = AnyEventObject>(
+  stateMachine: StateMachine.Machine<TC, TE, any>
+): {
+  state: Ref<StateMachine.State<TC, TE, any>>;
+  send: StateMachine.Service<TC, TE>['send'];
+  service: StateMachine.Service<TC, TE>;
+} {
+  const state = ref<StateMachine.State<TC, TE, any>>(stateMachine.initialState);
+  const service = interpret(stateMachine);
+  const send = (event: TE | TE['type']) => service.send(event);
+
+  service.subscribe(s => (state.value = s));
+  service.start();
```

Comment:
is this safe to do in here? I, unfortunately, don't have Vue expertise to judge that - but you have done the same thing for `useMachine` in `onBeforeMount`. Which one is more correct? I believe both should be started at the same time, right?

The edge-case scenario I'm concerned with is this:
1. render a component
2. while rendering change synchronously the state so the component in question gets dicarded

How far would Vue get with processing such a component? Would it be mounted (I suspect it wouldnt be)? And if it wouldnt get mounted - would `onBeforeUnmount` be called?

Reply:
Okay, I think I get now what you mean. I'm not also sure what would happen here, but I can try to construct a test case and evaluate that.

But generally for now we would be on the safe side if the services get initialized in `onMounted` and stopped on `onBeforeUnmout, right?

Reply:
Yes, that would certainly be safe and would align the behavior to what is implemented in `@xstate/react`. In React though it's the only way of setting up a subscription safely, if it's possible in Vue to start safely as service sooner then I wouldn't mind having a difference between those packages.